frontend/src/components/activePage/ActiveUtils.tsx

```diff
@@ -0,0 +1,265 @@
+  ApplyInstanceMutation,
+  Exact,
+  ItPolitoCrownlabsV1alpha2Instance,
+  UpdateType,
+  //useApplyInstanceMutation,
+} from '../../generated-types';
+import { getInstancePatchJson } from '../../graphql-components/utils';
+import { Instance, Template, WorkspaceRole } from '../../utils';
+
+/* const setInstanceRunning = (
+  running: boolean,
+  name: string,
+  tenantNamespace: string
+) => {
+  const [setInstanceRunning] = useApplyInstanceMutation();
+  return;
+}; */
+
+const getInstances = (
+  instance: ItPolitoCrownlabsV1alpha2Instance,
+  index: number,
+  userId: string,
+  tenantNamespace: string
+) => {
+  const { metadata, spec, status } = instance!;
+  const { environmentList, templateName } = spec
+    ?.templateCrownlabsPolitoItTemplateRef?.templateWrapper
+    ?.itPolitoCrownlabsV1alpha2Template?.spec! as any;
+  const environmentItem = environmentList[0];
+  return {
+    id: index,
+    name: metadata?.name,
+    gui: environmentItem.guiEnabled,
+    persistent: environmentItem.persistent,
+    idTemplate: spec?.templateCrownlabsPolitoItTemplateRef?.name!,
+    templatePrettyName: templateName,
+    ip: instance?.status?.ip,
+    status: instance?.status?.phase,
+    url: status?.url,
+    timeStamp: metadata?.creationTimestamp,
+    tenantId: userId,
+    tenantNamespace: tenantNamespace,
+    running: spec?.running,
+  } as Instance;
+};
+
+const getManagerInstances = (
+  instance: ItPolitoCrownlabsV1alpha2Instance,
+  index: number
+) => {
+  const { metadata, spec, status } = instance!;
+  const { tenantNamespace } = metadata! as any;
+  const { environmentList, templateName } = spec
+    ?.templateCrownlabsPolitoItTemplateRef?.templateWrapper
+    ?.itPolitoCrownlabsV1alpha2Template?.spec! as any;
+  const {
+    firstName,
+    lastName,
+  } = spec?.tenantCrownlabsPolitoItTenantRef?.tenantWrapper?.itPolitoCrownlabsV1alpha1Tenant?.spec!;
+  const { tenantId } = spec?.tenantCrownlabsPolitoItTenantRef as any;
+  return {
+    id: index,
+    name: metadata?.name,
+    gui: environmentList?.[0]?.guiEnabled,
+    persistent: environmentList?.[0]?.persistent,
+    idTemplate: spec?.templateCrownlabsPolitoItTemplateRef?.name!,
+    templatePrettyName: templateName,
+    ip: instance?.status?.ip,
+    status: instance?.status?.phase,
+    url: status?.url,
+    timeStamp: metadata?.creationTimestamp,
+    tenantId: tenantId,
+    tenantNamespace: tenantNamespace,
+    tenantDisplayName: `${firstName}\n${lastName}`,
+    workspaceId: spec?.templateCrownlabsPolitoItTemplateRef?.namespace,
+    running: spec?.running,
+  } as Instance;
+};
+
+const getTemplatesMapped = (instances: Instance[]) => {
+  return Array.from(new Set(instances?.map(i => i.templatePrettyName))).map(
+    t => {
+      const instancesFiltered = instances?.filter(
+        ({ templatePrettyName: tpn }) => tpn === t
+      );
+      const {
+        idTemplate,
+        gui,
+        persistent,
+        workspaceId,
+      } = instancesFiltered![0];
```

Comment:
```suggestion
      const [{
        idTemplate,
        gui,
        persistent,
        workspaceId,
      }] = instancesFiltered!;
```

Reply:
instanceFiltered is an array, i can't remove that '[0]'

Reply:
The syntax I wrote is meant to extract the first element of the array during an assignment :)